### PR TITLE
Add support for the cmake build to mono.proj.

### PR DESCRIPTION
**Pull request by vargaz**
Enable the cmake build by default on wasm/osx/android.

### CODE REVIEW

**Review by marek-safar — APPROVED** [review]

File: src/mono/cmake/configure.cmake [hunk 1]
```diff
@@ -74,6 +73,15 @@ if (NOT DARWIN)
   ac_check_funcs (getentropy)
 endif()
 
+find_package(Threads)
+# Needed to find pthread_ symbols
+set(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}")
```
Comment (akoeplinger) [inline src/mono/cmake/configure.cmake:78]: how does this work on WebAssembly?

File: src/mono/mono.proj [hunk 2]
```diff
@@ -854,34 +860,220 @@
...
+      <_MonoCCOption>CC="$(XcodeDir)/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang"</_MonoCCOption>
+      <_MonoCXXOption>CXX="$(XcodeDir)/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++"</_MonoCXXOption>
+      <_MonoRunInitCompiler>false</_MonoRunInitCompiler>
+      <_MonoCMakeSysroot Condition="'$(TargetsiOS)' == 'true' and '$(TargetsiOSSimulator)' != 'true'">$(XcodeDir)/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS$(iOSVersion).sdk</_MonoCMakeSysroot>
+      <_MonoCMakeSysroot Condition="'$(TargetsiOS)' == 'true' and '$(TargetsiOSSimulator)' == 'true'">$(XcodeDir)/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator$(iOSVersion).sdk</_MonoCMakeSysroot>
+      <_MonoCMakeSysroot Condition="'$(TargetstvOS)' == 'true' and '$(TargetstvOSSimulator)' != 'true'">$(XcodeDir)/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS$(tvOSVersion).sdk</_MonoCMakeSysroot>
+      <_MonoCMakeSysroot Condition="'$(TargetstvOS)' == 'true' and '$(TargetstvOSSimulator)' == 'true'">$(XcodeDir)/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator$(tvOSVersion).sdk</_MonoCMakeSysroot>
```
Comment (akoeplinger) [inline src/mono/mono.proj:889]: nit: I don't think we need this anymore and can let cmake figure out how to find Xcode/clang and the sysroots.

File: src/mono/mono.proj [hunk 3]
```diff
@@ -854,34 +860,220 @@
...
+      <_MonoCMakeVersionMin Condition="'$(TargetstvOS)' == 'true'">$(tvOSVersionMin)</_MonoCMakeVersionMin>
+    </PropertyGroup>
+    <ItemGroup Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">
+      <_MonoCMakeArgs Include="-DCMAKE_SYSTEM_NAME=$(_MonoCMakeSystemName)"/>
+      <_MonoCMakeArgs Include="-DCMAKE_OSX_DEPLOYMENT_TARGET=$(_MonoCMakeVersionMin)" />
+      <_MonoCMakeArgs Include="-DCMAKE_OSX_SYSROOT='$(_MonoCMakeSysroot)'" />
+      <_MonoCMakeArgs Condition="'$(Platform)' == 'x64'" Include="-DCMAKE_OSX_ARCHITECTURES=x86_64"/>
+      <_MonoCMakeArgs Condition="'$(Platform)' == 'x86'" Include="-DCMAKE_OSX_ARCHITECTURES=i386"/>
+      <_MonoCMakeArgs Condition="'$(Platform)' == 'arm64'" Include="-DCMAKE_OSX_ARCHITECTURES=arm64"/>
+      <_MonoCMakeArgs Condition="'$(Platform)' == 'arm'" Include="-DCMAKE_OSX_ARCHITECTURES=armv7s"/>
```
Comment (akoeplinger) [inline src/mono/mono.proj:903]: we support both armv7 and armv7s:

```suggestion
      <_MonoCMakeArgs Condition="'$(Platform)' == 'arm'" Include="-DCMAKE_OSX_ARCHITECTURES=&quot;armv7;armv7s&quot;"/>
```

**Review by akoeplinger — APPROVED** [review]
Looks good, left a few comments that can be addressed in a follow-up PR.

File: src/mono/cmake/configure.cmake (hunk 1 above)
Comment (vargaz) [reply src/mono/cmake/configure.cmake:78]: It doesn't do anything there.

**Pull request merged**